Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm

```diff
@@ -0,0 +1,207 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "WKPDFPageNumberIndicator.h"
+
+#if ENABLE(PDF_PAGE_NUMBER_INDICATOR)
+
+#import "PDFPluginIdentifier.h"
+#import "UIKitSPI.h"
+#import "WKWebViewIOS.h"
+#import <WebCore/GraphicsTypes.h>
+#import <WebCore/LocalizedStrings.h>
+#import <WebCore/PlatformCAFilters.h>
+#import <pal/spi/cocoa/QuartzCoreSPI.h>
+#import <wtf/Markable.h>
+#import <wtf/RetainPtr.h>
+#import <wtf/Seconds.h>
+#import <wtf/WeakObjCPtr.h>
+
+static constexpr CGFloat indicatorFontSize = 16;
+static constexpr CGFloat indicatorLabelOpacity = 0.4;
+static constexpr CGFloat indicatorCornerRadius = 7;
+static constexpr CGFloat indicatorMargin = 20;
+static constexpr CGFloat indicatorVerticalPadding = 6;
+static constexpr CGFloat indicatorHorizontalPadding = 10;
+
+static constexpr Seconds indicatorTimeout { 2_s };
+static constexpr Seconds indicatorFadeInDuration { 0.1_s };
+static constexpr Seconds indicatorFadeOutDuration { 0.75_s };
+static constexpr Seconds indicatorMoveDuration { 0.3_s };
+
+@implementation WKPDFPageNumberIndicator {
+    RetainPtr<UILabel> _label;
+    RetainPtr<UIVisualEffectView> _backdropView;
+    RetainPtr<NSTimer> _timer;
+    WeakObjCPtr<WKWebView> _webView;
+    Markable<WebKit::PDFPluginIdentifier> _pluginIdentifier;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame view:(WKWebView *)view pluginIdentifier:(WebKit::PDFPluginIdentifier)pluginIdentifier pageCount:(size_t)pageCount
+{
+    self = [super initWithFrame:frame];
+    if (!self)
+        return nil;
+
+    _pluginIdentifier = pluginIdentifier;
+    _webView = view;
+
+    [self setTranslatesAutoresizingMaskIntoConstraints:NO];
+    self.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
Is this necessary? I don't see any use of auto layout in this patch?

Reply:
not necessary, I will remove it